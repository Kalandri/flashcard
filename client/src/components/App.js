import React, { Component } from 'react';
import { connect } from 'react-redux';
import { BrowserRouter, Switch, Route } from 'react-router-dom';

import { fetchUser } from '../actions/index';
import Header from './Header';
import Dashboard from './Dashboard';
import Stack from './Stack';
import StackForm from './StackForm';
import Landing from './Landing';

import '../index.css';
import 'materialize-css/dist/css/materialize.min.css';
class App extends Component {
	componentDidMount(){
		this.props.fetchUser();
	}

	render(){
		return (
			<div >
			<BrowserRouter>
				<div>
					<Header />
				     <Switch>
				        <Route exact path='/' component={ Landing } />
				        <Route path='/dashboard' component = { Dashboard } />
				        <Route path='/stack' component={ Stack } />
				        <Route path='/stackform' component={ StackForm } />
				    </Switch>
		      	</div>
		    </BrowserRouter>
		    </div>
		);
	}
}

export default connect(null, { fetchUser })(App);