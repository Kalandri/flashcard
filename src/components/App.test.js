import React from 'react';
import { shallow } from 'enzyme';
import App from './App';

describe ('App', ()=>{
	const app = shallow(<App/>);
	it ('render the `Flashcard Pro` title', ()=>{
		expect(app.find('h2').text()).toEqual('Flashcard Pro');
	});

	it('renders the Stacklist', ()=>{
		expect(app.find('Connect(StackList)').exists()).toBe(true);
	});

	it('render a link to create anew stack', ()=>{
		expect(app.find('Link h4').text()).toEqual('Create a New Stack');
	});
})
