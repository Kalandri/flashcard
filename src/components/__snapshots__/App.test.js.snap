// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test for App 1`] = `
<div>
  <h2>
    Flashcard Pro
  </h2>
  <hr />
  <div>
    <a
      href="/stack"
      onClick={[Function]}
    >
      <h4>
        State Abbreviations
      </h4>
    </a>
    <a
      href="/stack"
      onClick={[Function]}
    >
      <h4>
        Computer Science Basics
      </h4>
    </a>
  </div>
  <hr />
  <a
    href="stack_form"
    onClick={[Function]}
  >
    <h4>
      Create a New Stack
    </h4>
  </a>
</div>
`;
